lib/compat.php

```diff
@@ -391,6 +399,8 @@ function gutenberg_render_block_with_assigned_block_context( $pre_render, $parse
 		return $pre_render;
 	}
 
+	$current_parsed_block = $parsed_block;
```

Comment:
We never know how this global will end up being used, a way to avoid having the global would be having a current_parsed_block private variable as part of WP_Block_Supports instance and a letter that allows setting the variable. Here instead of setting the variable, we would call that setter.